docs/react-types/ComponentProps.md

```diff
@@ -0,0 +1,57 @@
+---
+title: ComponentProps<T>
+---
+
+`ComponentProps<T>` is a utility type that lets you grab all valid props of an HTML or SVG element, or infer prop type for a component.
+
+## Parameters
+
+- `T extends keyof JSX.IntrinsicElements | JSXElementConstructor<any>`
```

Comment:
Let's just mention that you need to pass an element type not how this type is represented at the moment. `JSXElementConstructor` is a confusing name and should be replaced soon. 

Generally, let's focus on the type API/intended behavior not how we implement it at the moment.